airflow/models/taskinstance.py

```diff
@@ -1286,7 +1287,7 @@ def check_and_change_state_before_execution(
         return True
 
     def _date_or_empty(self, attr: str):
-        result = getattr(self, attr, None)  # type: datetime
+        result = getattr(self, attr, None)  # type: Optional[datetime]
```

Comment:
```suggestion
        result: Optional[datetime] = getattr(self, attr, None)
```

nit: Feels more cleaner

Reply:
I'll see if that works -- I just added the `Optional[]` without really thinking/examining too much.

Reply:
Yes, that works.